javascript/ql/src/semmle/javascript/frameworks/Vue.qll

```diff
@@ -366,4 +387,82 @@ module Vue {
   class VueFile extends File {
     VueFile() { getExtension() = "vue" }
   }
+
+  /**
+   * A taint propagating data flow edge through a Vue instance property.
+   */
+  class InstanceHeapStep extends TaintTracking::AdditionalTaintStep {
+
+    DataFlow::Node src;
+
+    InstanceHeapStep() {
+      exists(Instance i, string name, DataFlow::FunctionNode bound |
+        bound.flowsTo(i.getABoundFunction()) and
+        not bound.getFunction() instanceof ArrowFunctionExpr and
+        bound.getReceiver().getAPropertyRead(name) = this and
+        src = i.getAPropertyValue(name)
+      )
+    }
+
+    override predicate step(DataFlow::Node pred, DataFlow::Node succ) {
+      pred = src and succ = this
+    }
+
+  }
+
+  /*
+   * Provides classes for working with Vue templates.
+   */
+  module Template {
+    // Currently only supports HTML elements, but it may be possible to parse simple string templates later
+    private newtype TElement =
+      MkHtmlElement(HTML::Element e) { exists(VueFile f | e.getFile() = f) }
+
+    /**
+     * An element of a template.
+     */
+    abstract class Element extends TElement {
+      /** Gets a textual representation of this element. */
+      string toString() { result = "<" + getName() + ">...</>" }
+
+      /**
+       * Holds if this element is at the specified location.
+       * The location spans column `startcolumn` of line `startline` to
+       * column `endcolumn` of line `endline` in file `filepath`.
+       * For more information, see
+       * [LGTM locations](https://lgtm.com/help/ql/locations).
```

Comment:
As in https://github.com/Semmle/ql/pull/953, this should probably be changed to `help.semmle.com`.